examples/basic-infection/transmission_manager.rs

```diff
@@ -0,0 +1,61 @@
+use ixa::context::Context;
+use ixa::define_rng;
+use ixa::random::ContextRandomExt;
+
+use crate::people::ContextPeopleExt;
+use crate::people::InfectionStatus;
+use rand_distr::Exp;
+
+use crate::FOI;
+use crate::MAX_TIME;
+
+define_rng!(TransmissionRng);
+
+fn attempt_infection(context: &mut Context) {
+    let population_size: usize = context.get_population();
+    let person_to_infect: usize = context.sample_range(TransmissionRng, 0..population_size);
+
+    let person_status: InfectionStatus = context.get_person_status(person_to_infect);
+
+    if matches!(person_status, InfectionStatus::S) {
+        context.set_person_status(person_to_infect, InfectionStatus::I);
+        //context.schedule_recovery(person_to_infect);
+    }
+
+    #[allow(clippy::cast_precision_loss)]
+    let next_attempt_time = context.get_current_time()
```

Comment:
```suggestion
    // With a food-borne illness (i.e., constant force of infection),
    // each _person_ experiences an exponentially distributed
    // time until they fall sick because -- in this example --
    // all individuals are uniformly exposed. However, rather than
    // implementing this model by calculating each person's time to infection
    // at the beginning of the simulation and infecting them, we schedule an
    // infection time and _then_ go find the person to infect.
    // As a result, the FOI here is really a population-level FOI, and
    // we need to convert it to a per-person FOI to accurately model
    // a constant risk of disease to all people.
    let next_attempt_time = context.get_current_time()
```